src/core/language/language_common.cpp

```diff
@@ -88,6 +92,87 @@ static bool detectSetup()
 }
 
 
+void XLanguage::findLanguages()
+{
+    constexpr size_t prefix_length = 9; // std::string("thextech_").size();
+
+    // clear and free s_languages
+    {
+        std::vector<std::string> blank;
+        std::swap(s_languages, blank);
```

Comment:
What the problem with the `.clear()` ?

Reply:
I see (https://stackoverflow.com/questions/35514909/how-to-clear-vector-in-c-from-memory).
I guess, you would make a generic util call that can be used at other places too.

Reply:
Doesn't free the RAM, could be important if the number of languages shrinks significantly between calls. But in the ordinary cases, `clear()` should be fine. We could replace it for readability.